### DISCUSSION TITLE
Example of use

### DISCUSSION THREAD

**Discussion by papperkash** [alexkn/node-red-contrib-dlna#13] [Q&A] [answered]
Hi.

I have dragged the DLNA node in but am unable to get it to work, hoping for some help.

What would help would be a clear example of usage within node red. Perhaps a flow that can be imported to show a simple example?

I have tried to use the 'Inject' node to pass the following variables:

![image](https://user-images.githubusercontent.com/61649224/119114373-0627f800-ba1e-11eb-965d-318450c89289.png)

Unfortunately i just get this error: "TypeError: Cannot read property 'github' of undefined". If i change the msg.payload.url to a string then i get this: "Error: unsupported action: undefined"

**Comment by alexkn** [comment, marked as answer]
The right side should all be strings.
You have to use the "load" action to load a url.
Action "play" is only to unpause a paused stream.

**Comment by jmorris644** [comment]
It would be helpful to put this information in the MD file.

` <code>msg.payload.action</code> can be "play", "pause", "stop", "load", "getvolume", "setvolume".<br>
                For action "load" <code>msg.payload.url</code> must also be set.
                For action "setvolume" <code>msg.payload.volume</code> must also be set.`

**Comment by cogweg** [comment]
I could get in running to play radio stream like this with an node-RED "inject node":
msg.payload.action = "load"
msg.payload.url ="https://dispatcher.rndfnk.com/br/br2/sued/mp3/mid"      (this is the official mp3 stream of that radio station)
msg.device=  use the <friendlyname> of a upnp device description of the dlna player

I gathered the <friendlyname> with the Android App "UPNP Tool".  But usually, this value is also shown as name for a player in any UPNP Controller (e.g.BubbleUPNP).

**Reply by kisccc888** [reply]
It works after testing, thank you!

**Comment by kisccc888** [comment]
How do I set the volume?How to use it in node_red, thank you!
